tests/mysql/test-mysql.cpp

```diff
@@ -297,7 +297,7 @@ void test_num(const char* s, bool valid, T value)
         {
             char const * expectedPrefix = "Cannot convert data";
             CAPTURE(e.what());
-            CHECK(strncmp(e.what(), expectedPrefix, strlen(expectedPrefix)) == 0);
+            CHECK( (strncmp(e.what(), expectedPrefix, strlen(expectedPrefix)) == 0) );
```

Comment:
We have a few occurrences of `strncmp(s, foo, strlen(foo)) == 0` in the tests, I think it would make sense to add a helper `starts_with(s, foo)` function which could be used instead, this would make the code more clear and less error-prone and also avoid the problem with the parentheses (I'm surprised that it happens only here, what about the other places `strncmp(...) == 0` is used?)